utkdigitalinitiatives/TRACE
# Trac 1029a

** JIRA Ticket**:  
https://jira.lib.utk.edu/browse/TRAC-1022 [finding trace.utk.edu items with Google Scholar]
https://jira.lib.utk.edu/browse/TRAC-1029 [Create the Year-Index.htm page for trace.utk.edu]

* Other Relevant Link
https://scholar.google.com/intl/en/scholar/inclusion.html

# What does this Pull Request do?
This pull request creates a world-readable search-by-date page for Google Scholoar Inclusion.

# The files to be pulled are:

NEW:      /vagrant/files/islandora-solr-facet.tpl.php
REVISE:  /vagrant/files/islandora-solr-metadata-display.tpl.php

REVISE:  /vagrant/scripts/custom_scripts/163_revise_email_templates.sh  
NEW:      /vagrant/scripts/custom_scripts/164_revise_facet_template_for_date.sh
NEW:      /vagrant/scripts/custom_scripts/165_revise_metadata-display_template_for_date.sh

REVISE: /vagrant/scripts/custom_scripts/create_pages.php
                This was the most extensive revision.
               a. add IBY_BODY  code to build the HTML file presented for Google Scholar Inclusion
               b. revise HOME_BODY code to create the link (http://localhost:8000/iby) to give access to the IBY_BODY page.

# Purpose of this pull resuest:

Google Scholar Inclusion requires a world readable Search-by-date page that has links to 
  Search all collections by year.
  Search all collections for most recent two weeks.
  These requirements are met by the page http://localhost:8000/iby

All of the searches described below search the field:  mods_originInfo_encoding_edtf_keyDate_yes_dateModified_dt

If you think another field is better, we need to have the discussion immediately.

Google Scholar Inclusion requires a world readable Search-by-date page that has links to 
  Search all collections by year.
  Search all collections for most recent two weeks.
  These requirements are met by the page http://localhost:8000/iby which will become http://trace.utk.edu/iby

Adding the search-by-date facets in search results makes the site easier for users.

# How should this be tested?

The only Collection that works for submission on vagrant is Graduate Theses and Dissertations.

1. Checkout Trace(TRAC-1029a), do vagrant up.

2. On the home page, click on the link: "Find latest publications! Search by date across all collections."
   -- This should take you to:  http://localhost:8000/iby 
   -- If not, contact me immediately.  We have to deliver this for Google Scholar Inclusion.

3.  Add an etd by userA 
     Make sure the Date of Award is ***IN THE PAST*** or else the etd will not be seen in the collection until after the future date of award.
     SUBMIT.  LOGOUT.

     LOGIN as thesis_manager, accept, publish.

     LOG OUT and become a non-authenticated user because these searches are supposed to be World-Readable.

4.   Go back to localhost:8000/iby
     Search on "Last 10 minutes"
     This should return the etd userA just added. 

5.   Return to Browse Collections, Graduate Theses and Dissertations.
      Enter a YEAR into the Search Repository text box.
      This should bring up the date facets.
      (Searching on non-date terms will not bring up date facets.)
      From the date facets list, click on "Last 10 minutes".
      This should return the etd userA just added. 

Thank you for testing my pull request.

# Interested parties
@DonRichards 
@CanOfBees 
@utkdigitalinitiatives/digital-initiatives-developers
